File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max_len = 0;

        HashMap<Character,Integer>mpp = new HashMap<>();

        int left = 0;
        int right = 0;

        int n = s.length();

        while(right < n){

            char ch = s.charAt(right);

            if(mpp.containsKey(ch)){
                left = Math.max(left, mpp.get(ch) + 1);
            }
            
            mpp.put(ch,right);

            max_len = Math.max(max_len,right-left+1);
            right++;
        }

        return max_len;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {

        int n = s.size();
        unordered_map<char,int>mpp;

        int maxLen = 0;
        int left = 0;
        int right = 0;

        while(right < n){
    
            if(mpp.find(s[right]) != mpp.end()){
                left = max(left,mpp[s[right]] + 1);
            }

            mpp[s[right]] = right;

            maxLen = max(maxLen, right - left + 1);
            right++;
        }

        return maxLen;
    }
};